airbyte-cdk/python/airbyte_cdk/sources/file_based/config/abstract_file_based_spec.py

```diff
@@ -66,17 +66,32 @@ def replace_enum_allOf_and_anyOf(schema: Dict[str, Any]) -> Dict[str, Any]:
         for format in objects_to_check["oneOf"]:
             for key in format["properties"]:
                 object_property = format["properties"][key]
-                if "allOf" in object_property and "enum" in object_property["allOf"][0]:
-                    object_property["enum"] = object_property["allOf"][0]["enum"]
-                    object_property.pop("allOf")
+                AbstractFileBasedSpec.move_enum_to_root(object_property)
 
         properties_to_change = ["validation_policy"]
         for property_to_change in properties_to_change:
             property_object = schema["properties"]["streams"]["items"]["properties"][property_to_change]
             if "anyOf" in property_object:
                 schema["properties"]["streams"]["items"]["properties"][property_to_change]["type"] = "object"
                 schema["properties"]["streams"]["items"]["properties"][property_to_change]["oneOf"] = property_object.pop("anyOf")
-            if "allOf" in property_object and "enum" in property_object["allOf"][0]:
-                property_object["enum"] = property_object["allOf"][0]["enum"]
-                property_object.pop("allOf")
+            AbstractFileBasedSpec.move_enum_to_root(property_object)
+
+        csv_format_schemas = list(
```

Comment:
This is very ugly but I don't know enough about pydantic to understand how to do better. I would have like the `AbstractFileBasedSpec` not to know about the underlying objects but `CsvHeaderDefinition .schema` wasn't called. I guess this is why this method exists in the first place but it creates a pattern where `AbstractFileBasedSpec` knows too much